src/main/java/com/corndel/cashregister/Level4.java

```diff
@@ -9,7 +11,21 @@ public class Level4 {
    * the drawer. Returns false if it is not possible.
    */
   public static boolean canMakeAmount(int target, List<Item> drawer) {
-    // TODO
-    return false;
+    System.out.println(target);
+    
+    // Reverse order of list
+    Collections.reverse(drawer);
```

Comment:
Hey,

The reason why you are unable to see the print when it is below Collections.reverse() is because that line is throwing an exception.

In the test, the method is given a List.of(). This is an immutable/unmodifiable list. In order to reverse a list, it needs to be mutable.

If you add the following code to your method, you will be able to reverse the list.

```java
public class Level4 {
  /**
   * Returns true if it is possible to make the target amount out of the cash in
   * the drawer. Returns false if it is not possible.
   */
  public static boolean canMakeAmount(int target, List<Item> drawer) {
    System.out.println(target);

    List<Item> modifiableList = new ArrayList<>(drawer);

    // Reverse order of modifiableList
    Collections.reverse(modifiableList);

    System.out.println(target);
    
    // ... rest of code    
    
    }
}
```

You shouldn't need to reverse the list but now you should be able to 👍

Reply:
Thanks Charlie. Good to know that lists can be either immutable or mutable depending on how they are created.